### DISCUSSION TITLE
Property 'addEventListener' does not exist on type 'AbortSignal' inside [node]

### DISCUSSION THREAD

**Discussion by pixtron** [DefinitelyTyped/DefinitelyTyped#57805] [Issues with a @types package]
The interface `AbortSignal` misses `onabort`, `addEventListener`, `removeEventListener` methods.

For example this code taken from [Node.js docs](https://nodejs.org/api/globals.html#class-abortsignal) throws the error `Property 'addEventListener' does not exist on type 'AbortSignal'`.
 
```
const ac = new AbortController();

// Or the EventTarget API...
ac.signal.addEventListener('abort', (event) => {
  console.log(event.type);  // Prints 'abort'
}, { once: true });

ac.abort();
```

The interface `AbortSignal` in `@types/node` is declared [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2aae4db6bbe2e078c0cbc84cc9cd43cb1e01d8fd/types/node/globals.d.ts#L60)

There is a comment on line 45 of the above file
> region borrowed
// from https://github.com/microsoft/TypeScript/blob/38da7c600c83e7b31193a62495239a0fe478cb67/lib/lib.webworker.d.ts#L633 until moved to separate lib 

Can the borrowed part be updated?

[a comment by an automated account is not shown]

**Comment by mohsen1** [comment]
`AbortSignal` should extend `EventTarge`. Please feel free to submit a PR!

**Comment by e6c31d** [comment]
@mohsen1 `EventTarget` is also not part of `@types/node`.

**Reply by mohsen1** [reply]
You can use interface augmentation to extend it

**Comment by thw0rted** [comment]
EventTarget should be added: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/59001

but making it a global will cause conflicts in projects that use both `@types/node` and `lib.dom.d.ts`, because Node's implementation is not fully API-compatible with the DOM one.  In [a PR I'm putting together now](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59905), I address this by creating the interface in a module (I called it `node:dom-events`) from which it can safely be imported.  If that PR goes through I may write another that addresses this issue, but it's a big enough set of changes that some discussion will probably be required first.
